source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingZivid/HalconConnectUsingZivid.cpp

```diff
@@ -0,0 +1,103 @@
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename)
+{
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D toHalconPointCloud(const Zivid::PointCloud &pc)
+{
+	// Extract data from Zivid point cloud
+	const auto width = static_cast<long int>(pc.width());
+	const auto height = static_cast<long int>(pc.height());
+
+	std::vector<float> array1dX(width * height, 0.0);
+	std::vector<float> array1dY(width * height, 0.0);
+	std::vector<float> array1dZ(width * height, 0.0);
+	std::vector<float> array1dR(width * height, 0.0);
+	std::vector<float> array1dG(width * height, 0.0);
+	std::vector<float> array1dB(width * height, 0.0);
+
+	auto data = pc.dataPtr();
+	size_t idx = 0;
+	for (size_t i = 0; i < width; ++i)
+		for (size_t j = 0; j < height; ++j)
+		{
+			idx = width * j + i;
+			if (!isnan(data[idx].x))
+			{
+				array1dX[idx] = data[idx].x;
+				array1dY[idx] = data[idx].y;
+				array1dZ[idx] = data[idx].z;
+				array1dR[idx] = data[idx].red();
+				array1dG[idx] = data[idx].green();
+				array1dB[idx] = data[idx].blue();
+			}
+		}
+
+	// Construct HObjectModel3D from x,y,z data
+	HalconCpp::HImage X{ "real", width, height, array1dX.data() };
+	HalconCpp::HImage Y{ "real", width, height, array1dY.data() };
+	HalconCpp::HImage Z{ "real", width, height, array1dZ.data() };
+	HalconCpp::HObjectModel3D pcHalcon;
+	pcHalcon.XyzToObjectModel3d(X, Y, Z);
+
+	// Color the object based on R,G,B data
+	HalconCpp::HTuple Rtup{ array1dR.data(), width * height };
+	HalconCpp::HTuple Gtup{ array1dG.data(), width * height };
+	HalconCpp::HTuple Btup{ array1dB.data(), width * height };
+	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "red", "points", Rtup);
+	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "green", "points", Gtup);
+	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "blue", "points", Btup);
+
+	// Delete zero-points (not set due to nans in the Zivid point cloud)
+	pcHalcon = pcHalcon.SelectPointsObjectModel3d("point_coord_z", 0.01, 10000.0);
+
+	return pcHalcon;
+}
+
+int main()
+{
+	try
+	{
+		// Connect to camera
+		std::cout << "Connecting to camera" << std::endl;
+		Zivid::Application zivid;
+		auto camera = zivid.connectCamera();
+
+		camera << Zivid::Settings::Iris{ 20 }
+			<< Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 8333 } }
+			<< Zivid::Settings::Filters::Outlier::Enabled::yes
+			<< Zivid::Settings::Filters::Outlier::Threshold{ 5 };
+
+		std::cout << "Capturing frame" << std::endl;
+		auto frame = camera.capture();
+		Zivid::PointCloud pointCloud = frame.getPointCloud();
+
+		// Convert from Zivid to Halcon point cloud
+		HalconCpp::HObjectModel3D pcHalcon = toHalconPointCloud(pointCloud);
```

Comment:
`const auto` for both. Also I would rename them to pointCloudZivid and pointCloudHalcon or zividPointCloud and halonPointCloud so it is clear that it is 'the same' thing in different format.